average.js

```diff
@@ -1,6 +1,12 @@
-
 function average(numbers) {
-  return numbers.reduce((p, c)=> p + c, 0) / numbers.length;
+  var n =numbers.length
+  return numbers.reduce((p,c)=>{
+    if (isNaN(c))
+    {
+      n-=1
+      return p;
+    }
+    return p+c;
+  }, 0) / n;
 }
-
-module.exports = {average};
+module.exports = {average};
```

Comment:
Try a new test: The tests in `test/average.test.js` express the functionality of the `average` function.
Consider this situation: In IoT applications, analog sensors will report occasional outliers. 'Outliers' are values so small or so large, they aren't expected to happen. We need new functionality to ignore these outliers.
First add a unit-test in `test/average.test.js` to supply outliers and expect them to be ignored. Name the test appropriately. Then implement it in the code.
You'll need thresholds to recognize outliers: Think of sensing battery-temperature values and determine reasonable values.

Reply:
I wrote a unit test and also wrote a condition to rule out the any outlier values. I have considered any negative value or any value greater than 150 to be outlier.

Reply:
I have done the required changes.